src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -272,8 +272,7 @@ private void checkMethodCall(DetailAST methodCall) {
             objCalledOn = objCalledOn.getLastChild();
         }
         final DetailAST expr = methodCall.findFirstToken(TokenTypes.ELIST).getFirstChild();
-        if (isObjectValid(objCalledOn)
-                && containsOneArgument(methodCall)
+        if (containsOneArgument(methodCall)
                 && containsAllSafeTokens(expr)
                 && isCalledOnStringFieldOrVariable(objCalledOn)) {
```

Comment:
`isObjectValid` checks if what is being called on is not STRING_LITERAL or NEW. `isCalledOnStringFieldOrVariable` check if what is being called on is `String` class.

Since the latter method declares it must be X, the other method which says it must not be Y is overriden and this is why the mutation survives.